src/NUnitEngine/nunit.engine/TestEngine.cs

```diff
@@ -90,17 +90,21 @@ public void Initialize()
                 InternalTrace.Initialize(Path.Combine(WorkDirectory, logName), InternalTraceLevel);
             }
 
-            Services.Add(new SettingsService(true));
-            Services.Add(new DomainManager());
-            Services.Add(new ExtensionService());
-            Services.Add(new DriverService());
-            Services.Add(new RecentFilesService());
-            Services.Add(new ProjectService());
-            Services.Add(new RuntimeFrameworkService());
-            Services.Add(new DefaultTestRunnerFactory());
-            Services.Add(new TestAgency());
-            Services.Add(new ResultService());
-            Services.Add(new TestFilterService());
+            // If caller added services beforehand, we don't add any
+            if (Services.ServiceCount == 0)
```

Comment:
How about `Services.Added` to indicate they have been added but may or may not be initialized?